src/filters.ts

```diff
@@ -139,21 +139,28 @@ angular.module('3ema.filters', [])
                     // Unique
                     for (let possibleMention of result) {
                         let identity = possibleMention.substr(2, 8);
-                        let html;
-
+                        let mentionName;
+                        let cssClass;
                         if (identity === '@@@@@@@@') {
-                            html = '<span class="mention all">' + $translate.instant('messenger.ALL') + '</span>';
+                            mentionName = $translate.instant('messenger.ALL');
+                            cssClass = 'all';
+                        } else if (identity === webClientService.me.id) {
+                            mentionName = webClientService.me.displayName;
+                            cssClass = 'me';
                         } else {
                             const contact = webClientService.contacts.get(possibleMention.substr(2, 8));
                             if (contact !== null) {
-                                html = '<span class="mention contact">' + contact.displayName + '</span>';
+                                // Add identity to class for a simpler parsing
+                                cssClass = 'id ' + identity;
+                                mentionName = contact.displayName;
                             }
                         }
 
-                        if (html !== undefined) {
+                        if (mentionName !== undefined) {
                             text = text.replace(
                                 new RegExp(escapeRegExp(possibleMention), 'g'),
-                                html,
+                                '<span class="mention ' + cssClass + '"'
+                                    + ' text="@[' + identity + ']">' + mentionName + '</span>',
```

Comment:
We need to be careful about script injection here. Should be fine in this case though, since `identity` can only be a possible mention, so it must match the RegEx.

In the mentionName line breaks (`<br>` tags) are processed, but not things like `<script>`. Also, this is a value you set yourself on your phone, so not much danger :) We should still fix that some time (but not in this PR).

Reply:
+1